test/RoaveTest/SecurityAdvisories/VersionTest.php

```diff
@@ -239,6 +359,49 @@ public function nonEquivalentVersionProvider() : array
             ['1.0.0.1', '1.0.0'],
             ['2.0.1.1', '2.0.1'],
             ['2.0.1.0.0.0', '2.0.2'],
+            ['0.1-patch-3.4', '0.0-patch-3.4'],
+            ['1.0.1-patch-3.4', '1.0-patch-3.4'],
+            ['1-patch-3.4', '1.0.2-patch-3.4'],
+            ['1.0.0.1-patch-3.4', '1.0.0-patch-3.4'],
+            ['2.0.1.1-patch-3.4', '2.0.1-patch-3.4'],
+            ['2.0.1.0.0.0-patch-3.4', '2.0.2-patch-3.4'],
+            ['1-rc', '1-stable'],
+            ['1-stable', '1-beta'],
+            ['1-beta', '1-b.1'],
+            ['1-b', '1-alpha'],
+            ['1-alpha', '1-a.1'],
+            ['1-a.1', '1-patch'],
+            ['1-patch', '1-p.1'],
+            ['1-alpha.1', '1-alpha.1.2'],
```

Comment:
Woah, didn't know modifiers can have sub-versions...

Reply:
well, yes, from what I know this one is totally valid `1.1.1-alpha.beta.1.1.1.1+metadata1`